### ISSUE TITLE
ksh `set -A` not recognized as an assignment

### ISSUE THREAD

**Issue by sckendall2** [koalaman/shellcheck#3019]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2154
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#! /bin/ksh
set -A a
print ${#a[@]}
```

#### Here's what shellcheck currently says:
```
Line 3	SC2154: a is referenced but not assigned.
```
#### Here's what I wanted or expected to see:

No warning. In KornShell, `set -A a` creates the empty indexed array `a`.
